deserializers/src/main/java/com/nextdoor/bender/deserializer/json/GenericJsonDeserializer.java

```diff
@@ -36,21 +41,56 @@
   protected JsonParser parser;
   private final List<FieldConfig> nestedFieldConfigs;
   private String rootNodeOverridePath;
+  private Base64.Decoder base64decoder;
+  private ByteArrayOutputStream byteArrayOutputStream;
+  private final boolean performBase64DecodeAndUnzip;
 
   public GenericJsonDeserializer(List<FieldConfig> nestedFieldConfigs) {
-    this(nestedFieldConfigs, null);
+    this(nestedFieldConfigs, null, false);
   }
 
   public GenericJsonDeserializer(List<FieldConfig> nestedFieldConfigs,
-      String rootNodeOverridePath) {
+                                 String rootNodeOverridePath,
+                                 boolean performBase64DecodeAndUnzip) {
     this.nestedFieldConfigs = nestedFieldConfigs;
     this.rootNodeOverridePath = rootNodeOverridePath;
+    this.base64decoder = Base64.getDecoder();
+    this.byteArrayOutputStream = new ByteArrayOutputStream();
+    this.performBase64DecodeAndUnzip = performBase64DecodeAndUnzip;
+  }
+
+  public byte[] readGzipCompressedData(byte[] data) throws IOException {
+    GZIPInputStream gzipInputStream = new GZIPInputStream(new ByteArrayInputStream(data));
+    int res = 0;
+    byte[] buf = new byte[1024];
+    while (res >= 0) {
```

Comment:
Please replace this with `IOUtils.copy()` [1]. Also, both `GZIPInputStream` and `IOUtils.copy` allow setting the default buffer size. I think this is something we should expose because it has performance implications.

[1] https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/IOUtils.html#copy(java.io.InputStream,%20java.io.OutputStream,%20int)

Reply:
done!